src/strategies/spidStrategy.ts

```diff
@@ -129,4 +134,29 @@ const spidStrategy = async (
   );
 };
 
+/**
+ * Reads dates information in x509 certificate and logs remaining time to its expiration date.
+ * @param samlCert x509 certificate as string
+ */
+function logSamlCertExpiration(samlCert: string): void {
+  try {
+    const tempDir = fs.mkdtempSync("saml-cert");
+    fs.writeFileSync(`${tempDir}/samlCert.pem`, samlCert);
```

Comment:
this looks quite fragile and heavy, you can use something like [x509](https://www.npmjs.com/package/x509) that doesn't require forking a child process

Reply:
fixed in 1a3e026